.github/workflows/deploy-pack.yml

```
@@ -1,51 +1,55 @@
-# There is no manual way to call this out to run this on tags via UI.
-# See: https://github.community/t/workflow-dispatch-from-a-tag-in-actions-tab/130561
-on: workflow_dispatch
-name: Deploy - Framework
+name: Pack and nuget
+
+on:
+  push:
+    tags:
+      - '*'
 
 jobs:
-  check-if-tag:
-    name: Set Package Version
+  notify_pending_production_deploy:
     runs-on: ubuntu-latest
-    outputs:
-      version: ${{steps.deployment.outputs.version}}
     steps:
-      - name: Checkout
-        run: |
-          REPOSITORY="https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git"
-          BRANCH="${GITHUB_REF/#refs\/heads\//}"
-
-          git version
-          git clone --no-checkout ${REPOSITORY} .
-          git config --local gc.auto 0
-
-          git -c protocol.version=2 fetch --no-tags --prune --progress --depth=2 origin +${GITHUB_SHA}:refs/remotes/origin/${BRANCH}
-          git checkout --progress --force -B $BRANCH refs/remotes/origin/$BRANCH
-
-
-      - name: Set Variables
-        id: deployment
-        shell: bash
+      -
+        name: Submit pending deployment notification
         run: |
-          if [ $(git describe --exact-match --tags HEAD &> /dev/null; echo $?) == 0 ]; then
-            echo "::set-output name=VERSION::$(git describe --exact-match --tags HEAD)"
-          else
-            echo "fatal: no tag detected for HEAD. Workflow will now stop."
-            exit 128;
-          fi
+          export TITLE="Pending osu-framework Production Deployment: $GITHUB_REF_NAME"
+          export URL="https://github.com/ppy/osu-framework/actions/runs/$GITHUB_RUN_ID"
+          export DESCRIPTION="Docker image was built for tag $GITHUB_REF_NAME and awaiting approval for production deployment:
```

```suggestion
          export DESCRIPTION="Awaiting approval for building NuGet packages for tag $GITHUB_REF_NAME:
```

I'm going to assume that the "approval" part is achieved thanks to the `environment: production` definitions on the remaining jobs, so that seems to check out. But if that's how that works, that would mean that nothing has been built yet at the point of submitting a message asking for approval, as per suggestion above.

---

That's correct, yep.